src/PhpSpec/Util/MethodAnalyser.php

```diff
@@ -50,15 +50,47 @@ public function reflectionMethodIsEmpty(\ReflectionMethod $method)
      */
     private function getCodeBody(\ReflectionMethod $reflectionMethod)
     {
-        $reflectionClass = $reflectionMethod->getDeclaringClass();
+        $endLine = $reflectionMethod->getEndLine();
+        $startLine = $reflectionMethod->getStartLine();
+        $reflectionClass = $this->getDeclaringClass($reflectionMethod, $startLine, $endLine);
 
-        $length = $reflectionMethod->getEndLine() - $reflectionMethod->getStartLine();
+        $length = $endLine - $startLine;
         $lines = file($reflectionClass->getFileName());
-        $code = join(PHP_EOL, array_slice($lines, $reflectionMethod->getStartLine()-1, $length+1));
+        $code = join(PHP_EOL, array_slice($lines, $startLine -1, $length+1));
 
         return preg_replace('/.*function[^{]+{/s', '', $code);
     }
 
+    private function getDeclaringClass(\ReflectionMethod $reflectionMethod, $methodStartLine, $methodEndLine)
+    {
+        $reflectionClass = $reflectionMethod->getDeclaringClass();
+
+        $fileName = $reflectionMethod->getFileName();
+        $trait = $this->getDeclaringTrait($reflectionClass->getTraits(), $fileName, $methodStartLine, $methodEndLine);
+
+        return $trait === null ? $reflectionClass : $trait;
+    }
+
+    /**
+     * @param  \ReflectionClass[] $traits
+     * @param  string  $file
+     * @param  int $start
+     * @param  int $end
+     *
+     * @return null|\ReflectionClass
+     */
+    private function getDeclaringTrait(array $traits, $file, $start, $end)
+    {
+        foreach ($traits as $trait) {
+            if ($trait->getFileName() == $file && $trait->getStartLine() <= $start && $trait->getEndLine() >= $end) {
+                return $trait;
+            }
+            return $this->getDeclaringTrait($trait->getTraits(), $file, $start, $end);
```

Comment:
this is wrong. You are always returning something for the first trait in ``$traits``, meaning that multiple traits on the class are not supported